src/CommunityToolkit.Maui.Markup.UnitTests/TypedBindingExtensionsTests.cs

```diff
@@ -289,8 +290,142 @@ public void ValueSetOnOneWayWithNestedPathBinding(bool setContextFirst, bool isD
 		Assert.AreEqual(textColor, entry.GetValue(Entry.TextColorProperty));
 	}
 
+	[TestCase(false, false)]
+	[TestCase(false, true)]
+	[TestCase(true, false)]
+	[TestCase(true, true)]
+	public void ValueSetOnOneWayWithNestedPathBindingWithIValueConverter(bool setContextFirst, bool isDefault)
+	{
+		var label = new Label();
+		var bindingMode = isDefault ? BindingMode.Default : BindingMode.OneWay;
+
+		var viewmodel = new NestedViewModel
+		{
+			Model = new NestedViewModel
+			{
+				Model = new NestedViewModel()
+			}
+		};
+
+		var colorToHexRgbStringConverter = new ColorToHexRgbStringConverter();
+
+		if (setContextFirst)
+		{
+			label.BindingContext = viewmodel;
+			label.Bind<Label, NestedViewModel, Color?, string>(Label.TextProperty,
+							static (NestedViewModel vm) => vm.Model?.Model?.TextColor,
+							new (Func<NestedViewModel, object?>, string)[]
+							{
+								(vm => vm, nameof(NestedViewModel.Model)),
+								(vm => vm.Model, nameof(NestedViewModel.Model)),
+								(vm => vm.Model.Model, nameof(NestedViewModel.Model.TextColor))
+							},
+							static (NestedViewModel vm, Color? color) =>
+							{
+								if (vm.Model?.Model?.TextColor is not null && color is not null)
+								{
+									vm.Model.Model.TextColor = color;
+								}
+							},
+							bindingMode,
+							converter: colorToHexRgbStringConverter);
```

Comment:
It feels like this code could be reused or moved outside of the `if` statement. I can't see any difference with the other if unless I have missed something?